### PR TITLE
Limit Viewport

### LINKED ISSUE #3602: Limit Viewport
I have been hearing about the feature that developers want to limit the viewport for the end-user.
##### Sample use-case:

Create a tourist application for NYC, users are restricted to:
- move the map in NYC area 
- only allowed to zoom between 8 - 14 level 
- tilt is limited to only 20 degrees.  
##### API:

This will result in exposing a similar API as the old raster API:
- setScrollableAreaLimit() // definition ready
- setMinZoom() // implemented
- setMaxZoom() // implemented
- setMinTilt()
- setMaxTilt()

iOS is currently thinking about implementing https://github.com/mapbox/mapbox-gl-native/issues/2457, which will work similar to the rotation snap-back animation we currently have at very low zoom levels.

### PR DESCRIPTION
**Pull request by tobrun**
Android binding integration of https://github.com/mapbox/mapbox-gl-native/pull/8583.

ezgif com-video-to-gif 13

closes #3602

### CODE REVIEW

File: platform/android/src/native_map_view.cpp [hunk 1]
```diff
@@ -348,6 +349,14 @@ void NativeMapView::setStyleJson(jni::JNIEnv& env, jni::String json) {
     map->setStyleJSON(jni::Make<std::string>(env, json));
 }
 
+void NativeMapView::setLatLngBounds(jni::JNIEnv& env, jni::Object<mbgl::android::LatLngBounds> jBounds) {
+    if(jBounds==NULL){
```
Comment (ivovandongen) [inline platform/android/src/native_map_view.cpp]: nit: some spaces

File: platform/android/src/native_map_view.cpp [hunk 2]
```diff
@@ -348,6 +349,14 @@ void NativeMapView::setStyleJson(jni::JNIEnv& env, jni::String json) {
...
+        map->setLatLngBounds(mbgl::LatLngBounds::world());
+    }else{
```
Comment (ivovandongen) [inline platform/android/src/native_map_view.cpp]: Some more spaces

File: platform/android/src/native_map_view.hpp [hunk 3]
```diff
@@ -110,6 +111,8 @@ class NativeMapView : public View, public Backend {
 
     void setStyleJson(jni::JNIEnv&, jni::String);
 
+    void setLatLngBounds(jni::JNIEnv& env, jni::Object<mbgl::android::LatLngBounds>);
```
Comment (ivovandongen) [inline platform/android/src/native_map_view.hpp]: nit, remove `env`

**Comment by tobrun** [comment]
#8583 landed, ready for final review @ivovandongen?

**Pull request merged**